docs/cheaha/slurm/slurm_tutorial.md

```diff
@@ -300,6 +300,154 @@ $ sacct -j 27101430
 27101430_2.+     extern                 USER          1  COMPLETED      0:0
 ```
 
+#### Example 4.1: Slurm Array Job for Line-by-Line Word Count
+
+This SLURM job script processes a text file line by line using an array job. First, let us create a directory to create and store the input files:
+
+```bash
+$mkdir input_files
+
+$cd input_files
+```
+
+Next, copy and paste the following script into your terminal to generate five random text files, each containing a different number of randomly selected words. The script creates a file with 1 to 10 randomly generated lines, where each line contains 5 to 20 randomly selected words from `/usr/share/dict/words`. The output is saved in files named random_file_1.txt, random_file_2.txt, ..., random_file_5.txt, with each file containing a unique, randomly determined number of lines and words per line. The input files along with its path are tracked in `file_list.txt`.
+
+```bash
+for i in {1..5}; do
+    num_lines=$(($RANDOM % 10 + 1))  # Random number of lines (1-10)
+    for _ in $(seq 1 $num_lines); do
+        num_words=$(($RANDOM % 20 + 5))  # Random words per line (5-20)
+        shuf -n $num_words /usr/share/dict/words | paste -s -d " "
+    done > "random_file_$i.txt"
+done
+
+# Use globbing to get all generated files and save them to file_list.txt
+ls random_file_*.txt  > file_list.txt
+```
+
+Let us use these input files for [Example 4.1](#example-41-slurm-array-job-for-line-by-line-word-count) and [Example 4.2](#example-42-counting-words-in-multiple-files-using-a-slurm-job-array).
+
+Save the following Slurm script as `line_word_count.job`. This SLURM job ensures that each task in the job array processes a single line from an input file and counts the number of words in that line. You will have to create a directory named `logs` to redirect the output and error files using the command:
+
+```bash
+$mkdir logs
+```
+
+```bash linenums="1"
+#!/bin/bash
+#SBATCH --job-name=line_word_count  ### Name of the job
+#SBATCH --cpus-per-task=1           ### Number of Tasks per CPU
+#SBATCH --mem=2G                    ### Memory required, 2 gigabyte
+#SBATCH --partition=express         ### Cheaha Partition
+#SBATCH --time=00:10:00             ### Estimated Time of Completion, 10 minutes
+#SBATCH --output=logs/%x_%A_%a.out  ### Slurm Output file, %x is job name, %A is array job id, %a is array job index
+#SBATCH --error=logs/%x_%A_%a.err   ### Slurm Error file, %x is job name, %A is array job id, %a is array job index
+
+### Define the input file, for instance, random_file_2.txt
+INPUT_FILE="$HOME/input_files/random_file_2.txt"
+
+### Extract the line corresponding to the current Slurm array task ID
+LINE=$(sed -n "${SLURM_ARRAY_TASK_ID}p" "$INPUT_FILE")
+
+### Count words in the line and print the result for each task
+WORD_COUNT=$(echo "$LINE" | wc -w)
```

Comment:
Make it more explicit what `wc -w` is doing here, and what the pipe is doing. These can be brief descriptions.